RecoLuminosity/LumiProducer/test/analysis/plugins/PCCNTupler.h

```diff
@@ -0,0 +1,140 @@
+#ifndef PCCNTupler_h
+#define PCCNTupler_h
+
+/** \class PCCNTupler
+ * ----------------------------------------------------------------------
+ * PCCNTupler
+ * ---------
+ * Summary: The pixel clusters are summed per pixel module per lumi
+ *          lumi section.
+ *
+ * ----------------------------------------------------------------------
+ * Author:  Chris Palmer
+ * ----------------------------------------------------------------------
+ *
+ *
+ ************************************************************/
+
+#include <map>
+
+#include "DataFormats/TrackerRecHit2D/interface/SiPixelRecHitCollection.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "FWCore/Framework/interface/EDAnalyzer.h"
+#include "Geometry/TrackerGeometryBuilder/interface/PixelGeomDetUnit.h"
+#include "TrackingTools/TrackAssociator/interface/TrackDetectorAssociator.h"
+#include "HLTrigger/HLTcore/interface/HLTConfigProvider.h"
+#include "SimDataFormats/PileupSummaryInfo/interface/PileupSummaryInfo.h"
+
+#include "TObject.h"
+#include "TH1F.h"
+
+using namespace reco;
+
+class TObject;
+class TTree;
+class TH1D;
+class TFile;
+class RectangularPixelTopology;
+class DetId; 
+
+class PCCNTupler : public edm::EDAnalyzer {
```

Comment:
all new code should use multi-threading-aware modules as a base.
``` edm::one::EDAnalyzer<> ```
is probably the easiest here